pkg/serverstate/statetest/statetest.go

```diff
@@ -42,6 +45,20 @@ func Test(t *testing.T, f Factory, rf RestartFactory) {
 					name = name[idx+1:]
 				}
 
+				skip := false
+				for _, skipTest := range skipTests {
+					if name == skipTest {
+						skip = true
+						break
+					}
+				}
+				if skip {
+					t.Run(name, func(t *testing.T) {
+						t.Skip("Test is on the skip list - ignoring")
+					})
+					continue
```

Comment:
nit/suggestion: we could put a `Loop:` above the `range funcs` on line 42, then simplify this skip check to something like this:

```go
Loop:
  for _, tf := range funcs {
    // existing code
    // ...
    for _, skipTest := range skipTests {
      if name == skipTest {
        t.Run(name, func(t *testing.T) {
          t.Skip("Test is on the skip list - ignoring")
        })
        continue Loop
      }
    }
```

Just a suggestion though, take or leave as you will :)

Reply:
Ex: https://go.dev/play/p/QYk6_P15hdv

Reply:
Ha, I did think about that. I go back and forth on loop labels - they're convenient, but not my favorite for readability.